src/odemis/acq/move.py

```diff
@@ -125,14 +128,18 @@ def getCurrentGridLabel(current_pos, stage):
     grid1_pos = stage_md[model.MD_SAMPLE_CENTERS][POSITION_NAMES[GRID_1]]
     grid2_pos = stage_md[model.MD_SAMPLE_CENTERS][POSITION_NAMES[GRID_2]]
     if current_pos_label == SEM_IMAGING:
+        grid1_pos.update(stage_md[model.MD_FAV_SEM_POS_ACTIVE])
+        grid2_pos.update(stage_md[model.MD_FAV_SEM_POS_ACTIVE])
         distance_to_grid1 = _getDistance(current_pos, grid1_pos)
         distance_to_grid2 = _getDistance(current_pos, grid2_pos)
-        return GRID_2 if distance_to_grid1 > distance_to_grid2 else GRID_1 
+        return GRID_2 if distance_to_grid1 > distance_to_grid2 else GRID_1
     elif current_pos_label == FM_IMAGING:
+        grid1_pos.update(stage_md[model.MD_FAV_FM_POS_ACTIVE])
+        grid2_pos.update(stage_md[model.MD_FAV_FM_POS_ACTIVE])
```

Comment:
Sorry, I mistook. I now see that this code is here used to differentiate between grid1 and grid2 position. So we need to make sure that the gridN_pos constains the x,y,z values from SAMPLE_CENTERS. Therefore, that should also be done inverted, as above. BTW, you can do it before the `if`, to avoid duplication, in line 128.